magda-authorization-api/src/createApiRouter.ts

```diff
@@ -144,6 +177,114 @@ export default function createApiRouter(options: ApiRouterOptions) {
         res.end();
     });
 
+    /**
+     * @apiGroup Auth
+     * @api {get} /v0/auth/users/whoAllowDatasetOpeartion Get authorised users for dataset operation
+     * @apiDescription Returns a list users who can perform specified dataset operation
+     *
+     * @apiParam {String} datasetId Optional dataset id if you want to specify which dataset you want OPA makes decision on.
+     *  You must either supply an Id of the dataset or supply the dataset metadata in the request body
+     *
+     * @apiParam {String} operationUri  Mandatory The uri of the operation required
+     *
+     * @apiParamExample {json} Optional dataset metadata request data. Example:
+     * {
+     *      "publishingState": "draft",
+     *      "accessControl": {
+     *          "ownerId": "xxxxx-xxxx-xxxx-xxxx",
+     *          "orgUnitOwnerId": "xxxxx-xxxx-xxxx-xxxx",
+     *          "preAuthorisedPermissionIds": ["xxxxx-xxxx-xxxx-xxxx"]
+     *      }
+     * }
+     *
+     * @apiSuccessExample {json} 200
+     *    [{
+     *        "id":"...",
+     *        "displayName":"Fred Nerk",
+     *        "email":"fred.nerk@data61.csiro.au",
+     *        "photoURL":"...",
+     *        "source":"google",
+     *        "isAdmin": true
+     *    }]
+     *
+     * @apiErrorExample {json} 401/500
+     *    {
+     *      "isError": true,
+     *      "errorCode": 401, //--- or 500 depends on error type
+     *      "errorMessage": "Not authorized"
+     *    }
+     */
+    router.get(
```

Comment:
Also is it possible to generalise this to `record`s instead of just datasets?

Like say we wanted to make it so an organisation or something was owned by an org unit?

Reply:
It's now using GET method & query string parameters.
Re:`generalise this to records instead of just datasets`: I think we probably don't have a generic story to cover all records. Some records probably need a different story --- one example is `organisation` record --- see my comment here: https://github.com/magda-io/magda/issues/2247#issuecomment-501512939